### PR TITLE
feat: update actor_name assignment in decorator

### PR DESCRIPTION
**Pull request by z0z0r4**
Close https://github.com/Bogdanp/dramatiq/issues/640

### CODE REVIEW

**Comment by Bogdanp** [comment]
This isn't the right approach, since it would be a breaking change to any currently-deployed code. I would prefer it if we checked for duplicate actor names and raised an error instead.

**Comment by z0z0r4** [comment]
Indeed, I think checking for duplicate actor_name is necessary. But at the same time, it is reasonable to change the default value. We should support functions decorated with the same function name by default, right? This should also be a design mistake.

**Comment by Bogdanp** [comment]
I think it's more nuanced than a plain design mistake. At least, I remember thinking about this quite a bit when I made the original decision to use `__name__`. For example, even if we used `__qualname__` here, renaming the module in which an actor is defined would then potentially lead to a production bug (actors are looked up by name, and messages already enqueued for `mod1.foo` would not be able to find `mod2.foo`). Using `__qualname__` now is out of the question since it would break users' code for the same reason. So, the only good option is to keep using `__name__` and to error when there are duplicates.

**Comment by z0z0r4** [comment]
> actors are looked up by name, and messages already enqueued for mod1.foo would not be able to find mod2.foo

Will the production environment keep the queue unemptied and update dramatiq? I don't think so, but there is no evidence.

I will keep this pr until I open another to add the exception you said.

**Comment by z0z0r4** [comment]
@Bogdanp fixed

File: dramatiq/actor.py [hunk 1]
```diff
@@ -255,7 +258,7 @@ def actor(
     """
     def decorator(fn: Callable[..., Union[Awaitable[R], R]]) -> Actor[P, R]:
         nonlocal actor_name, broker
-        actor_name = actor_name or fn.__name__
+        actor_name = actor_name or fn.__qualname__
```
Comment (Bogdanp) [inline dramatiq/actor.py:261]: This change shouldn't be here, no?

**Review by Bogdanp — CHANGES_REQUESTED** [review]

File: dramatiq/actor.py (hunk 1 above)
Comment (z0z0r4) [reply dramatiq/actor.py:261]: yeee, sorry

**Comment by z0z0r4** [comment]
fixed

**Pull request merged**